src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.cs

```diff
@@ -408,5 +381,46 @@ public ValueTask<(Solution solution, bool updated)> TryUpdateWorkspaceCurrentSol
                 return tuple.solution;
             }
         }
+
+        private sealed class LazySolution : IAsyncDisposable, IDisposable
```

Comment:
this seems like a generally useful pattern.  perhaps could elevate to a common location?